File: drestaurant-libs/drestaurant-restaurant/src/main/java/com/drestaurant/restaurant/domain/RestaurantCommandHandler.java
package com.drestaurant.restaurant.domain;

import static org.axonframework.eventhandling.GenericEventMessage.asEventMessage;

import org.axonframework.commandhandling.CommandHandler;
import org.axonframework.commandhandling.model.Aggregate;
import org.axonframework.commandhandling.model.AggregateNotFoundException;
import org.axonframework.commandhandling.model.Repository;
import org.axonframework.eventhandling.EventBus;

class RestaurantCommandHandler {

	private Repository<Restaurant> repository;
	private EventBus eventBus;

	public RestaurantCommandHandler(Repository<Restaurant> repository, EventBus eventBus) {
		this.repository = repository;
		this.eventBus = eventBus;
	}

	@CommandHandler
	public void handle(ValidateOrderByRestaurantCommand command) {
		try {
			Aggregate<Restaurant> restaurantAggregate = repository.load(command.getRestaurantId());
			restaurantAggregate.execute(restaurant -> restaurant.validateOrder(command.getOrderId(), command.getLineItems(), command.getAuditEntry()));
		} catch (AggregateNotFoundException exception) {
			eventBus.publish(asEventMessage(new RestaurantNotFoundForOrderEvent(command.getRestaurantId(), command.getOrderId(), command.getAuditEntry())));
		}
	}
	
	

}

File: drestaurant-libs/drestaurant-restaurant/src/main/kotlin/com/drestaurant/restaurant/domain/RestaurantCommandHandler.kt
package com.drestaurant.restaurant.domain

import org.axonframework.commandhandling.CommandHandler
import org.axonframework.commandhandling.model.AggregateNotFoundException
import org.axonframework.commandhandling.model.Repository
import org.axonframework.eventhandling.EventBus
import org.axonframework.eventhandling.GenericEventMessage.asEventMessage

internal open class RestaurantCommandHandler(private val repository: Repository<Restaurant>, private val eventBus: EventBus) {

    @CommandHandler
    fun handle(command: ValidateOrderByRestaurantCommand) = try {
        repository.load(command.restaurantId).execute { it.validateOrder(command.orderId, command.lineItems, command.auditEntry) }
    } catch (exception: AggregateNotFoundException) {
        eventBus.publish(asEventMessage<Any>(RestaurantNotFoundForOrderEvent(command.restaurantId, command.orderId, command.auditEntry)))
    }
}
